From foo@baz Fri Jul 24 11:00:10 PDT 2009
Date: Fri, 24 Jul 2009 11:00:10 -0700
From: Greg Kroah-Hartman <gregkh@suse.de>
Subject: Staging: hv: blkvsc: fix up driver_data usage

From: Greg Kroah-Hartman <gregkh@suse.de>

driver_data is gone now from struct device, so use the proper functions
to access it instead.

Thanks to Bill Pemberton for pointing out this build error.

Cc: Bill Pemberton <wfp5p@viridian.itc.Virginia.EDU>
Cc: Hank Janssen <hjanssen@microsoft.com>
Cc: Haiyang Zhang <haiyangz@microsoft.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/hv/blkvsc_drv.c |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

--- a/drivers/staging/hv/blkvsc_drv.c
+++ b/drivers/staging/hv/blkvsc_drv.c
@@ -341,7 +341,7 @@ static int blkvsc_probe(struct device *d
 	blkdev->target = device_info.TargetId; // this identified the device 0 or 1
 	blkdev->path = device_info.PathId; // this identified the ide ctrl 0 or 1
 
-	device->driver_data = blkdev;
+	dev_set_drvdata(device, blkdev);
 
 	// Calculate the major and device num
 	if (blkdev->path == 0)
@@ -457,7 +457,7 @@ Cleanup:
 
 static void blkvsc_shutdown(struct device *device)
 {
-	struct block_device_context *blkdev = (struct block_device_context*)device->driver_data;
+	struct block_device_context *blkdev = dev_get_drvdata(device);
 	unsigned long flags;
 
 	if (!blkdev)
@@ -786,7 +786,7 @@ static int blkvsc_remove(struct device *
 
 	struct device_context *device_ctx = device_to_device_context(device);
 	DEVICE_OBJECT* device_obj = &device_ctx->device_obj;
-	struct block_device_context *blkdev = (struct block_device_context*)device->driver_data;
+	struct block_device_context *blkdev = dev_get_drvdata(device);
 	unsigned long flags;
 
 	DPRINT_ENTER(BLKVSC_DRV);
